From: Ubuntu Core Developers <ubuntu-devel-discuss@lists.ubuntu.com>
Date: Thu, 14 Feb 2019 18:59:15 -0800
Subject: pydoc-use-pager

# DP: pydoc: use the pager command if available.
# DP: pydoc: use the pager command if available.
---
 Lib/pydoc.py | 2 ++
 1 file changed, 2 insertions(+)

diff --git a/Lib/pydoc.py b/Lib/pydoc.py
index 5de6bf6..f57c4e5 100644
--- a/Lib/pydoc.py
+++ b/Lib/pydoc.py
@@ -1479,6 +1479,8 @@ def getpager():
         return plainpager
     if sys.platform == 'win32':
         return lambda text: tempfilepager(plain(text), 'more <')
+    if hasattr(os, 'system') and os.system('(pager) 2>/dev/null') == 0:
+        return lambda text: pipepager(text, 'pager')
     if hasattr(os, 'system') and os.system('(less) 2>/dev/null') == 0:
         return lambda text: pipepager(text, 'less')
 
